Laravel-Backpack/CRUD
# [4.2][WIP] Move pro items outside core (aka open-core split)

## WHY

Fixes https://github.com/Laravel-Backpack/CRUD/issues/4098

### BEFORE - What was wrong? What was happening before this PR?

All of our features were public-source, but under a restricted proprietary license. This was bad in two ways:
- bad for maintainers - devs could easily pirate Backpack, that's what 99.1% of them did (instead of paying for it);
- bad for OSS developers - devs could NOT use Backpack as a base for their open-source projects; I mean they could, but then they'd have to tell their users to buy a Backpack license which... was very inconvenient; their project, in the end, was not _truly_ open-source either, because it had a dependency with a restrictive license for commercial use (Backpack);

### AFTER - What is happening after this PR?

We've slit the features into two projects:
- Backpack\CRUD
     - we consider it the Backpack Core (might rename it later); 
     - will be released under an open-source license (MIT or smth);
     - will include all basic features you need to build an admin panel;
- Backpack\PRO
     - a bundle of features that will help you build more complex admin panels
     - closed-source;
     - can be installed with composer from our private repo;
     - includes these features - https://github.com/Laravel-Backpack/CRUD/issues/4098

## HOW

### How did you achieve that, in technical terms?

Created a new repo and moved what was needed there.

### Is it a breaking change or non-breaking change?

VERY breaking.

### How can we test the before & after?

If you install 4.2, you should NOT get all the features. Only after you add this to your `composer.json`:


```json
    "repositories": [
        {
            "type": "composer",
            "url": "https://repo.backpackforlaravel.com/"
        }
    ],
```

and run `composer require backpack/pro` should you be able to do all you could previously do in Backpack.
